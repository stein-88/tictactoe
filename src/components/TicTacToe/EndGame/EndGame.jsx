import React from 'react'
import champ from '@images/thechampion.jpg'
import placar from '@images/allwinners.jpg'
import Btn from '@generics/Btn'
import { ATIE } from '@constants/config'
import schema from './data'

const EndGame = ({ winner, restartGame }) => {
	if (!winner || !schema) return null
	const { thewinner, playagain } = schema
	return (
		<div className="row">
			<div className="col-lg-9 col-12 m-auto">
				<section className="w-100 text-center">
					<p className="h1">{winner === ATIE ? ATIE : `${thewinner} ${winner}`}</p>
					<section className="w-100 px-5">
						<img
						className="w-100 img-fluid rounded py-3 px-5"
						src={winner === ATIE ? placar : champ}
						data-test={winner === ATIE ? 'placarimg' : 'champimg'}
						alt=""
						/>
					</section>
					<Btn text={playagain} onClick={restartGame} />
				</section>
			</div>
		</div>
	)
}

export default EndGame
