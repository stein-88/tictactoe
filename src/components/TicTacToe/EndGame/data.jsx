const schemaendgame = {
  thewinner: 'The winner is ',
  playagain: 'Play again',
};

export default schemaendgame;
