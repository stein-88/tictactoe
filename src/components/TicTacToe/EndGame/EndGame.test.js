import React from 'react'
import renderer from 'react-test-renderer'
import EndGame from './EndGame'

describe('Test on EndGame Component', () => {
	test('Should not render without props', () => {
		const compEndGame = renderer.create(<EndGame />)
		const jsonEndGame = compEndGame.toJSON()
		expect(jsonEndGame).toBeNull()
	})
	test('Should render with props', () => {
		const compEndGame = renderer.create(<EndGame winner={'X'} />)
		const jsonEndGame = compEndGame.toJSON()
		expect(jsonEndGame).toMatchSnapshot()
	})
	test('Should render with props winner Tie the image have to be placarimg', () => {
		const compEndGame = renderer.create(<EndGame winner={'Tie'} />)
		const expectEnd = compEndGame.root.findAllByProps({ ['data-test']: "placarimg" })
		expect(expectEnd).toHaveLength(1)
	})
	test('Should render with props winner X or O the image to be champimg', () => {
		const compEndGameX = renderer.create(<EndGame winner={'X'} />)
		const compEndGameO = renderer.create(<EndGame winner={'O'} />)
		const expectEndX = compEndGameX.root.findAllByProps({ ['data-test']: "champimg" })
		const expectEndO = compEndGameO.root.findAllByProps({ ['data-test']: "champimg" })
		expect(expectEndX).toHaveLength(1)
		expect(expectEndO).toHaveLength(1)
	})
})