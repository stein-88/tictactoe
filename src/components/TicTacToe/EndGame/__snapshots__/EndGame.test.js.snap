// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on EndGame Component Should render with props 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-9 col-12 m-auto"
  >
    <section
      className="w-100 text-center"
    >
      <p
        className="h1"
      >
        The winner is  X
      </p>
      <section
        className="w-100 px-5"
      >
        <img
          alt=""
          className="w-100 img-fluid rounded py-3 px-5"
          data-test="champimg"
          src="test-file-stub"
        />
      </section>
    </section>
  </div>
</div>
`;
